### ISSUE TITLE
SOLVED: Lab 5.2 Test Case 9_12

### ISSUE THREAD

**Issue by davytan07** [nus-cs2030/2223-s2#507]
# Summary

Changed my implementation after I was unable to find the bug for 9_16 in my previous implementation (which involved using a buffer wait event). [Link to my original issue for 9_16](https://github.com/nus-cs2030/2223-s2/issues/398)

## Description:

Encountered this issue in 9_12 where I consistently have the opposite self-checkout counters serving the customers. Anyone knows what could possibly be triggering this error?

## Screenshots

<img width="672" alt="image" src="https://user-images.githubusercontent.com/45758052/226507433-3a19b5c4-d4bb-4dcd-9dab-d0275da10070.png">

**Comment by davytan07** [comment]
Found the issue, turns out that I accidentally used the wrong logic for choosing the self-check counter to serve the customer, I accidentally had it such that it would look for the self-check with the lowest next available time. Need to remember that the customer will just join the first available self-check (not necessarily the one with the lowest available time). I made a function to get the self-check with the lowest next available time for the updating of the next available time for the GROUP of self-check counters, and ended up using that function in the selection of self-check counter to serve.
